x/liquidstaking/keeper/liquidstaking.go

```diff
@@ -1308,122 +1270,61 @@ func (k Keeper) handlePairedChunk(ctx sdk.Context, chunk types.Chunk) {
 					// there was double sign slashing after re-pairing, so in this case
 					// unpairing insurance doesn't have to pay for penalty
 				} else {
-					// TODO: Impelment rest of logics
-					coveredAmt, _, damagedChunk := k.mustCoverDoubleSignPenaltyFromUnpairingInsurance(ctx, chunk)
+					coveredAmt := k.mustCoverDoubleSignPenaltyFromUnpairingInsurance(ctx, chunk)
```

Comment:
1. **This case must be occurred when `chunk.UnpairingInsurance.Validator == chunk.PairedInsurance.Validator`**

2. above empty `if epoch.GetStartHeight() < latestEvidence.GetHeight() {` branch could be broken on lint

3. and above switch statement need to add `types.ErrInvalidValidatorStatus` case or add validation on the front of `handlePairedChunk`

Reply:
Updated it in [this commit](https://github.com/b-harvest/Canto/pull/9/commits/57fee4ba046390d1028e48117c9327e40a49eba4).

* Q1 is checked in https://github.com/b-harvest/Canto/blob/zsystm/simplify-penalty-logic/x/liquidstaking/keeper/liquidstaking.go#L1620-L1630
* For Q3, I add another case statement and panics because`ErrInvalidValidatorStatus` must not happen.
  * `ErrInvalidValidatorStatus` is returned if
    * validator status == unspecified or shares is nil or (tokens == 0 but shares > 0)
    * only extream cases like direct editing staking module's states will trigger those conditions.